### PR TITLE
Add errors to `ensure` functions  in `Base` of widgets

### PR DESCRIPTION
**Pull request by zm711**
This came up during talking in #2782, but I think `TypeError` instead of `ValueError` no? @alejoe91 ? 

And was there a reason you didn't want an error here originally @samuelgarcia @alejoe91 ?

### CODE REVIEW

**Review by alejoe91 — APPROVED** [review]

**Comment by zm711** [comment]
I see what the error was. The correlograms were the exception that needed to allow the sorting to pass through. I think this solution is a bit better since it checks for sorting and if not sorting then we do the ensure which will raise an error if someone passes something random. This should be more robust to end-users potentially passing the wrong thing (in my opinion :) )

File: src/spikeinterface/widgets/crosscorrelograms.py [hunk 1]
```diff
@@ -46,7 +46,9 @@ def __init__(
         backend=None,
         **backend_kwargs,
     ):
-        sorting_analyzer_or_sorting = self.ensure_sorting_analyzer(sorting_analyzer_or_sorting)
+
+        if not isinstance(sorting_analyzer_or_sorting, BaseSorting):
+            sorting_analyzer_or_sorting = self.ensure_sorting_analyzer(sorting_analyzer_or_sorting)
```
Comment (samuelgarcia) [inline src/spikeinterface/widgets/crosscorrelograms.py:51]: Here you break the fact that sorting can be an input no ?

**Comment by samuelgarcia** [comment]
Theses function were more here for automatic convertion (with MockWaveform) than for a script type checking.
Sometimes a hack or the type happen after.
The idea of not having an error was to be able to both Sorting or Analyzer for instance.
I think you broke this for correlogram.
I am not sure to valide 100%. Let me have a deeper look.

File: src/spikeinterface/widgets/crosscorrelograms.py (hunk 1 above)
Comment (zm711) [reply src/spikeinterface/widgets/crosscorrelograms.py:51]: Actually tests are passing now. 

The logic for the if statement is 

if base sorting => continue as base sorting
if not base sorting => ensure sorting analyzer (for sorting analyzer or mock waveform)

So both cases are addressed.

**Comment by alejoe91** [comment]
This looks good to me

File: src/spikeinterface/widgets/crosscorrelograms.py (hunk 1 above)
Comment (samuelgarcia) [reply src/spikeinterface/widgets/crosscorrelograms.py:51]: ok
then it is ok.

**Comment by samuelgarcia** [comment]
lets merge and see.

**Pull request merged**